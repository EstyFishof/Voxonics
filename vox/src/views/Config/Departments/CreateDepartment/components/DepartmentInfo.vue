<template>
  <div class="personal-info-wrap">
    <div class="wrap-head">
      <h2>DEPARTMENT INFO</h2>
      <span class="hr"></span>
    </div>

    <div class="wrap-body">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <label>
              Name
              <input placeholder="Enter name of Department"
                     class="input-field"
                     v-model.trim="name"
                     maxlength="12"
                     @change="setDepartmentName"
                     required
                     :class="{error: requiredName}"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"

export default {
  name: "DepartmentInfo",
  props: ['requiredName'],
  data() {
    return {
      name: "",
    }
  },
  computed: {
    ...mapGetters(['newDepartmentName'])
  },
  methods: {
    setDepartmentName() {
      this.$store.dispatch('transferNameDepartment', this.name)
    }
  },
  mounted() {
    if (this.newDepartmentName) {
      this.name = this.newDepartmentName
    }
  }
}
</script>

<style scoped lang="scss">
.personal-info-wrap {
  .wrap-body {
    display: flex;
    flex-direction: row;
    margin: 36px 0 36px 24px;

    .personal-info {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;

      &__first-part {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;

        .item-container {
          margin: 16px 0;

          label {
            font-size: 18px;
            font-weight: 700;
            text-align: left;
            color: var(--light-blue);

            .error {
              border: 1px solid red !important;
              outline: red;
            }
          }
        }
      }
    }
  }

  .with-err {
    position: relative;
    margin-bottom: 30px;
  }

}
</style>
