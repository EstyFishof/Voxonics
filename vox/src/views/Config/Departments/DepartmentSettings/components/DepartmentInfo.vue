<template>
  <div class="personal-info-wrap" :class="{'disabled-events': !$access.isEditDepAndQueue(userInfo, 'department', departmentInfo)}">
    <div class="wrap-head">
      <h2>DEPARTMENT INFO</h2>
      <span class="hr"></span>
      <span>
        <svg width="30" height="39" viewBox="0 0 30 39" fill="none" xmlns="http://www.w3.org/2000/svg" content="Delete department" v-tippy="{ placement : 'top',  arrow: true, delay : 800, duration: 500 }" @click="del()" class="trash" :class="{'grayscale': !$access.isEditDepAndQueue(userInfo, 'department', departmentInfo)}">
          <path d="M2.14286 34.6667C2.14286 37.05 4.07143 39 6.42857 39H23.5714C25.9286 39 27.8571 37.05 27.8571 34.6667V8.66667H2.14286V34.6667ZM30 2.16667H22.5L20.3571 0H9.64286L7.5 2.16667H0V6.5H30V2.16667Z" fill="#276899"/>
          <rect x="10" y="1" width="10" height="1" />
          <rect x="6" y="12" width="2" height="23" />
          <rect x="14" y="12" width="2" height="23" />
          <rect x="22" y="12" width="2" height="23" />
        </svg>
      </span>
    </div>

    <div class="wrap-body">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <span>  Name of Department:</span>
          <div class="item-container with-err">
            <div class="left-section-block">
              <input content="Change name" v-tippy="{ placement : 'top',  arrow: true, delay : 800, duration: 500 }"
                :style="{width: computedWidth}"
                    placeholder="Enter name of Department"
                    :class="{'input-field': edit, 'grayscale': !$access.isEditDepAndQueue(userInfo, 'department', departmentInfo)}"
                    :defaulValue="departmentInfoName"
                    v-model.trim="name"
                    maxlength="12"
                    @input="setDepartmentName"
              />
              <span class="last-update-block">
                <span class="last-update-text">Last updated:</span>
                <span class="last-update-data">{{departmentInfo.dateUpdated}}</span>
              </span>
            </div>
            <div class="wrap-static">
              <div class="static">
                <div class="label">Numbers in Department:</div>
                <div class="num">{{ numbersDep.length || 0 }}</div>
              </div>
              <div class="static">
                <div class="label">Agents in Department:</div>
                <div class="num">{{ agentsDep.length || 0 }}</div>
              </div>
              <div class="static">
                <div class="label">Managers in Department:</div>
                <div class="num">{{ managersDep.length || 0 }}</div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
import store from "../../../../../store"

export default {
  name: "DepartmentInfo",
  data() {
    return {
      name: "",
      edit: false
    }
  },
  computed: {
    ...mapGetters(['numbersDep', 'agentsDep', 'managersDep', 'departmentInfoName', 'userInfo', 'departmentInfo']),
    computedWidth() {
      return this.name.length ? 32 + this.name.length * 1.1 * 24 + 'px' : '100%'
    }
  },
  methods: {
    del() {
      this.$store.dispatch('setDeleteDep', true)
    },
    setDepartmentName() {
      this.$store.dispatch('changeNameDepartment', this.name)
    },
  },
  mounted() {
    this.name = this.departmentInfoName
    store.subscribe((mutation) => {
      if (mutation.type === 'pushDepartment') {
        this.name = this.departmentInfoName
      }
    })

  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/departmentsAndQueues/info.scss"></style>
<style scoped lang="scss">
.personal-info-wrap {
  .wrap-head {
    .trash{
      position: static;
      margin-left: 30px;
      margin-bottom: 10px;
      &:hover {
        transform: scale(1.2, 1.2);
      }
      rect {
        fill: var(--substrate);
      }
    }
  }
  .wrap-body {
    .personal-info {
      &__first-part {
        .item-container {
          input {
            margin-right: 100px;
          }
          .wrap-static {
            .static {
              margin-right: 50px;
            }
          }
        }
      }
    }
  }
}
</style>
