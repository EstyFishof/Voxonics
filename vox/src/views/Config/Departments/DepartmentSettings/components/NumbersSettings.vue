<template>
  <div class="personal-info-wrap" :class="{'disabled-events': !$access.isEditDepAndQueue(userInfo, 'department', departmentInfo)}">
    <div class="wrap-head">
      <h2>Numbers in Department</h2>
    </div>
    <div class="top">
      <div class="left" :class="{'disabled-gray-options-block': !$access.isEditDepAndQueue(userInfo, 'department', departmentInfo)}">Assign number
      </div>
      <div class="numbers">
        <div v-if="numbersDep.length" class="number">
          <div class="block">
            <div class="round">№ <span class="vnumber">VNumber</span></div>
            <span class="name"></span>
          </div>
          <span class="name"></span>
          <span class="union"></span>
        </div>
        <div v-if="numbersDep.length > 1" class="number">
          <div class="block">
            <div class="round">№ <span class="vnumber">VNumber</span></div>
            <span class="name"></span>
          </div>
          <span class="name"></span>
          <span class="union"></span>
        </div>
        <div v-if="numbersDep.length > 2" class="number">
          <div class="block">
            <div class="round">№ <span class="vnumber">VNumber</span></div>
            <span class="name"></span>
          </div>
          <span class="name"></span>
          <span class="union"></span>
        </div>
        <div v-if="numbersDep.length > 3" class="number">
          <div class="block">
            <div class="round">№ <span class="vnumber">VNumber</span></div>
            <span class="name"></span>
          </div>
          <span class="name"></span>
          <span class="union"></span>
        </div>
      </div>
    </div>
    <div class="wrap-body" :class="{'disabled-gray-options-block': !$access.isEditDepAndQueue(userInfo, 'department', departmentInfo)}">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <div class="label" v-click-outside="hideListNumbers">
              <input placeholder="Enter or select number"
                     class="input-field"
                     @click="showList()"
                     @input="showNumbers = true; searchNumbers()"
                     v-model.trim="name"/>
              <span @click="showList()" class="triangle" :class="{'rotate-arrow': showNumbers}">
                <img src="@/assets/img/icons/select-arrow.svg" alt="select">
              </span>
              <DropdownList v-show="showNumbers" :classNameList="'list'" :dataList="mainList.getMainList"  @scrollFunc="(event) => funScroll(event)"
                  @clickFunc="(item) => chooseNumber(item)" :itemText="(item) => `<span>${item.description} ${item.number}</span>`"
                  :classNameItemComputed="(item) => numbersDep.filter(element => element.uid === item.uid).length ? 'selected row' : 'row'" />

            </div>
          </div>
        </div>
      </div>
      <div class="numbers">
        <div class="number" v-for="(number, index) of numbersDep" :key="index">
          <div class="block">
            <div class="round">{{ index + 1 }}.</div>
            <span class="name">{{ number.number }}</span>
            <span class="description">{{ number.description }}</span>
          </div>
          <span class="name">{{ number.internalNumber }}</span>
          <img @click="chooseNumber(number)" class="union" src="@/assets/img/icons/union.svg" width="19" alt="">
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
import store from "../../../../../store"
import ClickOutside from "vue-click-outside"
import DropdownList from '@/components/DropdownList'
import {dropdownListFetch, dropdownChooseItem} from '@/lib/dropdownListsDepsAndQueues.js'
import _ from 'lodash'

export default {
  name: "VNumbersSetting",
  components: {
    DropdownList
  },
  data() {
    return {
      unsubscribeStore: () => {
      },
      name: "",
      showNumbers: false,
      perPage: 12,
      mainList: {
        topList: [],
        bottomList: [],
        getMainList: []
      },
    }
  },
  watch: {
    getGlobalExternalNumbers() {
      dropdownListFetch(this, this.getGlobalExternalNumbers, 'numbersDep')
      this.mainList.getMainList = _.concat(this.mainList.topList, this.mainList.bottomList)
    }
  },
  computed: {
    ...mapGetters(['numbersDep', 'userInfo', 'departmentInfo', 'getGlobalExternalNumbers', 'getPaginatorNumbers']),
  },
  methods: {
    searchNumbers() {
      clearTimeout(this.timer)
      this.timer = setTimeout(() => this.fetch(), 800)
    },
    funScroll({target}) {
      if((target.clientHeight + target.scrollTop === target.scrollHeight)) {
        if(this.getPaginatorNumbers.paginator.count > this.perPage) {
          this.perPage += 9
          this.fetch()
        }
      }
    },
    fetch() {
      this.$store.dispatch('setGlobalExternalNumbers', {peerPage: this.perPage, number: this.name})
    },
    showList() {
      this.fetch()
      this.showNumbers = !this.showNumbers
    },
    hideListNumbers() {
      this.showNumbers = false
    },
    chooseNumber(numProp) {
      const num = _.omit(numProp, 'usedIn')
      dropdownChooseItem(this, num, 'numbersDep', 'setNumbersDepartment')
    },
    setNumbersDepartment(array) {
      this.$store.dispatch('changeNumbersDepartment', array)
    }
  },
  mounted() {
    this.popupItem = this.$el.children[2].children[0]
    this.unsubscribeStore = store.subscribe((mutation) => {
      if (mutation.type === 'pushDepartment') {
        this.mainList.topList = JSON.parse(JSON.stringify(this.numbersDep))
      }
    })
    this.mainList.topList = JSON.parse(JSON.stringify(this.numbersDep))
  },
  beforeDestroy() {
    this.unsubscribeStore()
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/departmentsAndQueues/agentsSettings.scss"></style>
<style scoped lang="scss">
.personal-info-wrap {
  width: 100%;
  .top {
    .number {
      .vnumber {
        margin-left: 20px;
      }
    }
  }

  .numbers {
    .number {
      .block {
        position: relative;

        .round {
          border-radius: 0;
          margin-right: 0;
          border: none;
        }

        .description {
          position: absolute;
          left: 25px;
          top: 21px;
          font-weight: 300;
          font-size: 10px;
          line-height: 9px;
          color: #8F8F8F;
        }
      }
    }
  }
}
</style>
