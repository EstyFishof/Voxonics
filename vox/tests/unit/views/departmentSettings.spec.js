import {
    shallowMount,
    createLocalVue
} from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import DepartmentSettings from '../../../src/views/Config/Departments/DepartmentSettings/DepartmentSettings.vue';
import AgentsSetting from '../../../src/views/Config/Departments/DepartmentSettings/components/AgentsSettings.vue';
import DepartmentInfo from '../../../src/views/Config/Departments/DepartmentSettings/components/DepartmentInfo.vue';
import ManagersSetting from '../../../src/views/Config/Departments/DepartmentSettings/components/ManagersSettings.vue';
import NumbersSettings from '../../../src/views/Config/Departments/DepartmentSettings/components/NumbersSettings.vue';
import VueTippy, { TippyComponent } from "vue-tippy";


const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);
const router = new VueRouter();
localVue.use(Vuex);

describe('DepartmentSettings', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            department: {
                name: "",
                relatedWith: {
                    externalNumber: [],
                    user: []
                },
                belongTo: { user: [] }
            },
        },
        mutations: {
            depInfoMutation(state, value) {
                state.department = value;
            }
        },
        getters: {
            departmentInfo(state) {
                return state.department;
            },
        }
    });
    const wrapper = shallowMount(DepartmentSettings, {
        store,
        localVue,
        router,
        stubs: {DepartmentSettings: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    // it('Check snapshot', () => {
    //     expect(wrapper.element).toMatchSnapshot();
    // })
    it('setShowOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setShowOptions');
        await wrapper.vm.setShowOptions('config');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('handlerSubmit toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'handlerSubmit');
        await wrapper.vm.handlerSubmit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('clearDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'clearDepartment');
        await wrapper.vm.clearDepartment();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('mounted component', () => {
        DepartmentSettings.mounted.call(wrapper.vm);
    });
    it('mounted component', async () => {
        store.commit('depInfoMutation', {
            uid: 1111
        });
        await DepartmentSettings.mounted.call(wrapper.vm);
    });
});

describe('AgentsSetting', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            agents: [],
            depAgents: [],
        },
        actions: {
            changeAgentsDepartment({
                commit
            }, value) {}
        },
        mutations: {
            queueInfoMutation(state, value) {
                state.queueInfo = value;
            }
        },
        getters: {
            agents(state) {
                return state.agents;
            },
            agentsDep(state) {
                return state.depAgents;
            },
        }
    });
    const wrapper = shallowMount(AgentsSetting, {
        store,
        localVue,
        router,
        stubs: {AgentsSetting: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        await wrapper.vm.chooseAgent({
            uid: 111
        });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        wrapper.setData({
            relatedAgents: [{
                uid: 111
            }]
        });
        await wrapper.vm.chooseAgent({
            uid: 111
        });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setAgentsDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setAgentsDepartment');
        await wrapper.vm.setAgentsDepartment();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('computed agentsFiltered', () => {
    //     AgentsSetting.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({
    //         name: '123'
    //     })
    //     AgentsSetting.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({
    //         name: 'aaa'
    //     })
    //     AgentsSetting.computed.agentsFiltered.call(wrapper.vm)
    // })
    it('mounted component', async () => {
        await AgentsSetting.mounted.call(wrapper.vm);
    });
});

describe('DepartmentInfo', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            depNumbers: [],
            depAgents: [],
            depManagers: [],
            name: ''
        },
        actions: {
            changeNameDepartment({
                commit
            }, value) {},
            setDeleteDep({commit}, value) {}
        },
        getters: {
            numbersDep(state) {
                return state.depNumbers;
            },
            agentsDep(state) {
                return state.depAgents;
            },
            managersDep(state) {
                return state.depManagers;
            },
            departmentInfoName(state) {
                return state.name;
            },
        }
    });
    const wrapper = shallowMount(DepartmentInfo, {
        store,
        localVue,
        router,
        stubs: {DepartmentInfo: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('del toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'del');
        await wrapper.vm.del();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setDepartmentName toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setDepartmentName');
        await wrapper.vm.setDepartmentName();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('toggleEdit toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'toggleEdit');
        await wrapper.vm.toggleEdit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('computed agentsFiltered', () => {
        DepartmentInfo.computed.computedWidth.call(wrapper.vm);
    });
    it('mounted component', async () => {
        await DepartmentInfo.mounted.call(wrapper.vm);
    });
});

describe('ManagersSetting', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            depManagers: [],
            managerDepInfo: {
                user: [
                    {
                        uid: '',
                        option: {
                            edit: 1
                        }
                    }
                ],
            },
            managersDep: [
                {
                    avatar: '',
                    name: '',
                    internalNumber: '',
                    option: {
                        edit: false
                    }
                }
            ]
        },
        actions: {
            changeManagersDepartment({
                commit
            }, value) {},
        },
        getters: {
            managers(state) {
                return state.depManagers;
            },
            managersDep(state) {
                return state.managerDepInfo;
            },
        }
    });
    const wrapper = shallowMount(ManagersSetting, {
        store,
        localVue,
        router,
        stubs: {ManagersSetting: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        wrapper.setData({relatedManagers: [{uid: 222, option: {edit: true}}]});
        await wrapper.vm.chooseAgent({uid: 111, option: {edit: true}});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        wrapper.setData({relatedManagers: [{uid: 111, option: {edit: true}}]});
        await wrapper.vm.chooseAgent({uid: 111, option: {edit: true}});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('changeEdit toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeEdit');
        await wrapper.vm.changeEdit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setAgentsDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setAgentsDepartment');
        await wrapper.vm.setAgentsDepartment();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('agentsFiltered agentsFiltered', async () => {
        await ManagersSetting.computed.agentsFiltered.call(wrapper.vm);
    });
    it('agentsFiltered agentsFiltered', async () => {
        wrapper.setData({name: '123'});
        await ManagersSetting.computed.agentsFiltered.call(wrapper.vm);
    });
    it('agentsFiltered agentsFiltered', async () => {
        wrapper.setData({name: 'aaa'});
        await ManagersSetting.computed.agentsFiltered.call(wrapper.vm);
    });
    it('mounted component', async () => {
        await wrapper.setData({relatedManagers: [{option: {edit: true}}]});
        await ManagersSetting.mounted.call(wrapper.vm);
    });
});

describe('NumbersSettings', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            depNumbers: [],
            numberDepInfo: {
                uid: "a27cb70c-2560-481e-8039-f0e10d48bbc0",
                number: "380443640000",
                name: "Work phone",
                country: "UA kiev",
                network: "test"
            }
        },
        actions: {
            changeNumbersDepartment({
                commit
            }, value) {},
        },
        getters: {
            numbers(state) {
                return state.depNumbers;
            },
            numbersDep(state) {
                return state.numberDepInfo;
            },
        }
    });
    const wrapper = shallowMount(NumbersSettings, {
        store,
        localVue,
        router,
        stubs: {NumbersSettings: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('chooseNumber toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseNumber');
        wrapper.setData({relatedNumbers: [{uid: 222}]});
        await wrapper.vm.chooseNumber({uid: 111});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseNumber toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseNumber');
        wrapper.setData({relatedNumbers: [{uid: 222}]});
        await wrapper.vm.chooseNumber({uid: 111});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setNumbersDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setNumbersDepartment');
        await wrapper.vm.setNumbersDepartment();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('numbersFiltered computed', async () => {
    //     await NumbersSettings.computed.numbersFiltered.call(wrapper.vm)
    // })
    // it('numbersFiltered computed', async () => {
    //     wrapper.setData({name: '123'})
    //     await NumbersSettings.computed.numbersFiltered.call(wrapper.vm)
    // })
    // it('numbersFiltered computed', async () => {
    //     wrapper.setData({name: 'aaa'})
    //     await NumbersSettings.computed.numbersFiltered.call(wrapper.vm)
    // })
    it('mounted component', async () => {
        await NumbersSettings.mounted.call(wrapper.vm);
    });
});
