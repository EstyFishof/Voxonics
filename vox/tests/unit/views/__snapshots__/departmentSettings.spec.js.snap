// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentsSetting Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      Agents in Department
    </h2>
  </div>
   
  <div
    class="top"
  >
    <div
      class="left"
    >
      Add agent
    
    </div>
     
    <div
      class="agents users"
    >
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <div
            class="label"
          >
            <input
              class="input-field"
              placeholder="Enter name or Agent Number"
            />
             
            <span
              class="list"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="agents users"
    />
  </div>
</div>
`;

exports[`DepartmentInfo Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      DEPARTMENT INFO
    </h2>
     
    <span
      class="hr"
    />
     
    <img
      alt="trash"
      class="trash"
      content="Delete department"
      src="@/assets/img/icons/trash.svg"
      tabindex="0"
      width="30"
    />
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <span>
            Name of Department:
        </span>
         
        <div
          class="item-container with-err"
        >
          <input
            class=""
            content="Change name"
            defaulvalue=""
            maxlength="12"
            placeholder="Enter name of Department"
            style="width: 100%;"
          />
           
          <div
            class="wrap-static"
          >
            <div
              class="static"
            >
              <div
                class="label"
              >
                Numbers in Department:
              </div>
               
              <div
                class="num"
              >
                0
              </div>
            </div>
             
            <div
              class="static"
            >
              <div
                class="label"
              >
                Agents in Department:
              </div>
               
              <div
                class="num"
              >
                0
              </div>
            </div>
             
            <div
              class="static"
            >
              <div
                class="label"
              >
                Managers in Department:
              </div>
               
              <div
                class="num"
              >
                0
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ManagersSetting Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      Managed by
    </h2>
  </div>
   
  <div
    class="top"
  >
    <div
      class="left"
    >
      Add manager
      
    </div>
     
    <div
      class="managers users"
    >
      <!---->
       
      <!---->
    </div>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <div
            class="label"
          >
            <input
              class="input-field"
              placeholder="Enter name or Agent Number"
            />
             
            <span
              class="list"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="managers users"
    />
  </div>
</div>
`;

exports[`NumbersSettings Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      Numbers in Department
    </h2>
  </div>
   
  <div
    class="top"
  >
    <div
      class="left"
    >
      Assign number
      
    </div>
     
    <div
      class="numbers"
    >
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <div
            class="label"
          >
            <input
              class="input-field"
              placeholder="Enter or select number"
            />
             
            <span
              class="list"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="numbers"
    >
      <div
        class="number"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            uid1.
          </div>
           
          <span
            class="name"
          >
            
          </span>
           
          <span
            class="description"
          >
            
          </span>
        </div>
         
        <span
          class="name"
        >
          
        </span>
         
        <img
          alt=""
          class="union"
          src="@/assets/img/icons/union.svg"
          width="19"
        />
      </div>
      <div
        class="number"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            number1.
          </div>
           
          <span
            class="name"
          >
            
          </span>
           
          <span
            class="description"
          >
            
          </span>
        </div>
         
        <span
          class="name"
        >
          
        </span>
         
        <img
          alt=""
          class="union"
          src="@/assets/img/icons/union.svg"
          width="19"
        />
      </div>
      <div
        class="number"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            name1.
          </div>
           
          <span
            class="name"
          >
            
          </span>
           
          <span
            class="description"
          >
            
          </span>
        </div>
         
        <span
          class="name"
        >
          
        </span>
         
        <img
          alt=""
          class="union"
          src="@/assets/img/icons/union.svg"
          width="19"
        />
      </div>
      <div
        class="number"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            country1.
          </div>
           
          <span
            class="name"
          >
            
          </span>
           
          <span
            class="description"
          >
            
          </span>
        </div>
         
        <span
          class="name"
        >
          
        </span>
         
        <img
          alt=""
          class="union"
          src="@/assets/img/icons/union.svg"
          width="19"
        />
      </div>
      <div
        class="number"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            network1.
          </div>
           
          <span
            class="name"
          >
            
          </span>
           
          <span
            class="description"
          >
            
          </span>
        </div>
         
        <span
          class="name"
        >
          
        </span>
         
        <img
          alt=""
          class="union"
          src="@/assets/img/icons/union.svg"
          width="19"
        />
      </div>
    </div>
  </div>
</div>
`;
